lib/sidekiq/processor.rb

```diff
@@ -79,7 +83,10 @@ def run
 
     def process_one(&block)
       @job = fetch
-      process(@job) if @job
+      if @job
+        @job.lifecycle = self if @job.is_a?(Job::Iterable)
```

Comment:
Here, job will always be a `UnitOfWork` so this will never execute. You want to set this right before middleware is called, right after `JobClass.new`.

Reply:
Right 🤦 The unit tests are too unit (with mocking), weren't able to detect this.